core/jazz_es-kinesis-log-streamer/components/utils.js

```diff
@@ -136,96 +202,236 @@ function getApiLogsData(payload) {
+                  let messageLength = Buffer.byteLength(message, 'utf8');
+                  if (messageLength > 32766) { //since 32766(32KB) is the default message size
+                    let truncatedMessage = truncate(message, 32740); // message size + ...[TRUNCATED]
+                    data.message = truncatedMessage + "  ...[TRUNCATED]";
+                  } else {
+                    data.message = message;
+                  }
+
+                  data.log_level = getSubInfo(logEvent.message, config.PATTERNS.log_level, 0);
+                  if (!data.log_level) {
+                    data.log_level = config.DEFAULT_LOG_LEVEL;
+                  }
+
+                  if (!(data.message.startsWith("REPORT") || data.message.startsWith("START") || data.message.startsWith("END"))) {
+                    let timestmp = getSubInfo(data.message, config.PATTERNS.timestamp_pattern, 0);
+                    data.message = data.message.replace(timestmp, "");
+
+                    let guid = getSubInfo(data.message, config.PATTERNS.guid_regex, 0);
+                    data.message = data.message.replace(guid, "");
+
+                    data.message = data.message.replace(data.log_level, "");
+                  }
+                  data.message = data.message.trim();
+                  let indexName = "applicationlogs";
+                  let action = {
+                    "index": {}
+                  };
+                  action.index._index = indexName;
+                  action.index._type = "applicationlogs";
+                  action.index._id = logEvent.id;
+
+                  bulkRequestBody += [
+                    JSON.stringify(action),
+                    JSON.stringify(data),
+                  ].join('\n') + '\n';
+                });
+              } else {
+                logger.error("invalid lambda logs event..: " + JSON.stringify(payload));
+              }
+              callback(bulkRequestBody);
+            } else {
+              callback(null);
+            }
+          })
+        })
+      })
+    })
+  })
+}
 
-        data.platform_log_group = payload.logGroup;
-        data.platform_log_stream = payload.logStream;
-        data.timestamp = new Date(1 * logEvent.timestamp).toISOString();
-        let message = logEvent.message;
-        let messageLength = Buffer.byteLength(message, 'utf8');
-        if (messageLength > 32766) { //since 32766(32KB) is the default message size
-          let truncatedMessage = truncate(message, 32740); // message size + ...[TRUNCATED]
-          data.message = truncatedMessage + "  ...[TRUNCATED]";
+// Function to get service metadata using service API
+function getsServiceMetaData(config, logGroup, authToken) {
+  var serviceParts = logGroup.split('_');
+  return new Promise((resolve, reject) => {
+    var service_api_options = {
+      url: `${config.SERVICE_API_URL}${config.SERVICE_URL}?domain=${serviceParts[1]}&service=${serviceParts[2]}&environment=${serviceParts[4]}`,
```

Comment:
This will break when user supplied function names contain `_` (we don't restrict how developers can name their functions). Instead, look for the last part, if it is equal to `dev`, environment_id = last two parts, if not the last part.

Reply:
Refactored the code. For environment, now we are accessing from the extreme right side